autofill/autofill-api/src/main/java/com/duckduckgo/autofill/AutofillCredentialDialogs.kt

```diff
@@ -22,7 +22,7 @@ import com.duckduckgo.autofill.domain.app.LoginCredentials
 /**
  * Dialog which can be shown when user is required to select which saved credential to autofill
  */
-interface CredentialAutofillPickerDialog : DialogFragmentType {
+interface CredentialAutofillPickerDialog {
```

Comment:
Not for this PR but maybe for a future one. Looking at these interfaces I think they can probably be deleted altogether and move the companions inside the implementations. The exception is the usage in `AutofillCredentialsSelectionResultHandler` which lives in the `app` module and could also be moved to `impl` to avoid a few roundtrips between the module and the tab fragment.

Reply:
Agree not for this PR, and agree this could be improved. Would be good to define a guideline around this scenario first, then come back and make the changes based on that guideline.

Reply:
Will track and discuss more here: [Autofill: Move more of autofill dialogs into autofill-impl module](https://app.asana.com/0/1202727816093912/1202727816093912/f)